README.md

```diff
@@ -24,39 +26,63 @@ To install Go, visit [this link](https://golang.org/dl/).
 ## Usage
 Before using MSAL Go, you will need to [register your application with the Microsoft identity platform](https://docs.microsoft.com/azure/active-directory/develop/quickstart-v2-register-an-app).
 
+### Public Surface
+
+The Public API of the library can be found in the following directories under `apps`.
+
+```
+apps/ - Contains all our code
+  confidential/ - The confidential application API
+  public/ - The public application API
+  cache/ - The cache interface that can be implemented to provide persistence cache storage of credentials
+```
+
 Acquiring tokens with MSAL Go follows this general three step pattern. There might be some slight differences for other token acquisition flows. Here is a basic example:
+
 1. MSAL separates [public and confidential client applications](https://tools.ietf.org/html/rfc6749#section-2.1). So, you would create an instance of a `PublicClientApplication` and `ConfidentialClientApplication` and use this throughout the lifetime of your application.
-```go
-publicClientApp, err := msal.CreatePublicClientApplication("your_client_id",
-    "https://login.microsoftonline.com/Enter_the_Tenant_Name_Here")
-```
+   * Initializing a public client:
 
-2. MSAL comes packaged with an in-memory cache. Utilizing the cache is optional, but we would highly recommend it.
-```go
-accounts := publicClientApp.GetAccounts()
-// Assuming the user wanted the first account
-userAccount := accounts[0]
-scopes := []string{"your_scope"}
-silentParams := msal.CreateAcquireTokenSilentParametersWithAccount(scopes, userAccount)
-result, err := publicClientApp.AcquireTokenSilent(silentParams)
-```
+    ```go
+    app, err := public.New("client_id", public.WithAuthority("https://login.microsoftonline.com/Enter_The_Tenant_Name_Here"))
+    ```
+
+   * Initializing a confidential client:
+
+    ```go
+    // Initializing the client credential
+    cred, err := confidential.NewCredFromSecret("client_secret")
```

Comment:
>NewCredFromSecret [](start = 30, length = 17)

this name is odd, should be as it is in other libraries. New is fine but CredFromSecret sounds wrong. It's the ConfidentialClient, can also remove this and stick to simply New.

Reply:
`NewCredFromSecret` is a publicAPI [func](https://github.com/AzureAD/microsoft-authentication-library-for-go/blob/dev/apps/confidential/confidential.go#L147) to create the credential object. The initialization of the application still uses `confidential.New()`

Reply:
thanks for clarifying, I suppose I object to not sticking more to the MSAL naming to have an easier transition if you know one of our other MSAL's.